### PR TITLE
feat: publish event when user email changed

### PR DESCRIPTION
**Pull request by Judge40**
When a user account is updated to change the email, an event should be published including the original and new email addresses.

TIS21-6006
TIS21-6009
TIS21-6010

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ application:
     cognito:
       user-pool-id: ${COGNITO_USER_POOL_ID}
       dsp-consultant-group: ${COGNITO_DSP_CONSULTANT_GROUP}
+    sns:
+      user-account:
+        update: ${USER_ACCOUNT_UPDATE_EVENT_TOPIC:}
```
Comment (ReubenRobertsHEE) [inline src/main/resources/application.yml:17]: I see that permissions to push to this SNS have already been added ✔️

File: src/main/java/uk/nhs/tis/trainee/usermanagement/service/EventPublishService.java [hunk 2]
```diff
@@ -54,6 +66,26 @@ public void publishSingleProfileSyncEvent(String traineeTisId) {
...
+   */
+  public void publishEmailUpdateEvent(String userId, String traineeId, String previousEmail,
+      String newEmail) {
+    log.info("Publishing email update event for previous email '{}' and new email '{}'.",
+        previousEmail, newEmail);
+    EmailUpdateEvent event = new EmailUpdateEvent(userId, traineeId, previousEmail, newEmail);
+    notificationMessagingTemplate.convertAndSend(userAccountUpdateTopicArn, event, Map.of(
+        NOTIFICATION_SUBJECT_HEADER, "Account Email Updated",
+        MESSAGE_GROUP_ID_HEADER, userId,
+        "producer", "tis-trainee-user-management"
```
Comment (ReubenRobertsHEE) [inline src/main/java/uk/nhs/tis/trainee/usermanagement/service/EventPublishService.java:88]: This is interesting. Are you keen to see anything published to SNS including this sort of information so we know where it originally came from? I haven't noticed us using these headers before.

File: src/main/resources/application.yml (hunk 1 above)
Comment (Judge40) [reply src/main/resources/application.yml:17]: I remembered this time, there's a first time for everything 😆

File: src/main/java/uk/nhs/tis/trainee/usermanagement/service/EventPublishService.java (hunk 2 above)
Comment (Judge40) [reply src/main/java/uk/nhs/tis/trainee/usermanagement/service/EventPublishService.java:88]: It's a bit of a trial off the back of my realisation it's not possible to identify the queue a DLQ message came from. If we can at least identify where it originated we may have a better chance.

**Review by ReubenRobertsHEE — APPROVED** [review]

**Pull request merged**